### PR TITLE
Generate validator interface for middleware

### PR DESCRIPTION
**Pull request by shiiyan**
## Description

Closes https://github.com/sivchari/govalid/issues/77

This pull request introduces a major enhancement to the generated validation code by implementing a standard `Validator` interface for all validated types. This change makes it easier to use these types with generic code and middleware, and improves integration with frameworks (e.g., HTTP handlers). The update includes changes to code templates, documentation, and golden test files.

Detailed explanations are documented in README.
https://github.com/sivchari/govalid/blob/09878af2c3767ef44cf0a3c156f563baf25e32c7/README.md#32-validator-interface

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [x] Code passes `make golangci-lint` without errors
- [x] All tests pass: `make test`
- [x] Code follows existing patterns and conventions
- [x] No hard-coded values (use constants or configuration)
- [x] Error messages are clear and actionable

### Testing Requirements
- [x] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [x] Unit tests implemented with boundary value testing (`test/unit/`)
- [x] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [ ] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [go-ozzo/ozzo-validation](https://github.com/go-ozzo/ozzo-validation) (BenchmarkOzzoValidation*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [x] All test files generated: `cd test && go generate`

### Performance Standards
- [ ] Zero allocations in validation logic (verified via benchmarks)
- [ ] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [x] README.md updated with new marker documentation
- [x] Code includes appropriate comments and examples
- [x] Validator usage examples provided

### Integration
- [ ] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [ ] Registry files automatically updated
- [x] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [x] All GitHub Actions CI checks pass
- [x] No breaking changes to existing functionality
- [x] Contribution follows project license requirements

## Related Issues

https://github.com/sivchari/govalid/issues/77

### CODE REVIEW

**Comment by sivchari** [comment]
Please fix the lint

File: validation/middleware/middleware_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			handlerCalled := false
+			testHandler := func(w http.ResponseWriter, r *http.Request) {
+				handlerCalled = true
+				person := r.Context().Value(middleware.ValidatedBodyKey).(*fixture.PersonRequest)
+				w.Write([]byte("Person created: " + person.Name))
+			}
```
Comment (sivchari) [inline validation/middleware/middleware_test.go:54]: Maybe, it's okay for this handler to just return http.StatusOK.

File: validation/middleware/middleware_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/sivchari/govalid/validation/middleware/fixture"
+)
+
+func TestValidateRequest(t *testing.T) {
+	tests := []struct {
+		name           string
+		requestBody    interface{}
+		expectedStatus int
+		expectedBody   string
+		handlerCalled  bool
```
Comment (sivchari) [inline validation/middleware/middleware_test.go:20]:
```suggestion
        name string
		want int
```

File: validation/middleware/middleware_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	}{
+		{
+			name:           "valid request",
+			requestBody:    fixture.PersonRequest{Name: "John", Email: "john@example.com"},
+			expectedStatus: http.StatusOK,
+			expectedBody:   "Person created: John",
+			handlerCalled:  true,
+		},
```
Comment (sivchari) [inline validation/middleware/middleware_test.go:28]:
```suggestion
		{
			name:           "valid request",
			want: http.StatusOK,
		},
```

**Review by sivchari — COMMENTED** [review]
I left some comments.

File: validation/middleware/middleware.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/sivchari/govalid"
+)
+
+type ContextKey string
+
+const ValidatedBodyKey ContextKey = "validatedBody"
```
[the remark this answers is not in the document]
Comment (shiiyan) [reply validation/middleware/middleware.go:13]: Yes, I agree with your opinion. The original idea was to save the validated struct into the request context so handlers or use cases could reuse it without decoding JSON again. However, in real-world scenarios, users might want to use request models generated by OpenAPI or other tools. So I think it would be better to let the user choose how they want to use the request body.

**Comment by shiiyan** [comment]
@sivchari 

Thank you for your kind comments. I will take some time to work on them in recent days.

File: validation/middleware/middleware.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	return func(w http.ResponseWriter, r *http.Request) {
+		var body T
+
+		if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+			http.Error(w, "Invalid JSON", http.StatusBadRequest)
+			return
+		}
+
+		if err := body.Validate(); err != nil {
+			http.Error(w, "Validation error: "+err.Error(), http.StatusBadRequest)
```
[the remark this answers is not in the document]
Comment (sivchari) [reply validation/middleware/middleware.go:23]: Make sense to me. Sorry for misleading.

File: Makefile [hunk 6]
```diff
@@ -50,6 +50,7 @@ generate-validator: ## Generate a new validator scaffold and all registry files.
 test: ## Run all tests except validation helper (due to known issues)
 	go test ./... -shuffle on -v -race
 	go test -C test ./... -shuffle on -v -race
+	go test ./... -shuffle on -v -race -tags=middleware_test
```
[the remark this answers is not in the document]
Comment (shiiyan) [reply Makefile]: https://github.com/sivchari/govalid/pull/136/commits/0d75afe49eb79e00dfe92dafbf65dbd6658d06e5

Fixed here. Please have a look.

**Review by sivchari — APPROVED** [review]
Thx!
